### PR TITLE
docs: review checkbox.mdx

### PR DESCRIPTION
**Pull request by julia-rabello**
#### What is the purpose of this pull request?
Reviewing **Checkbox** component documentation and adding design guidelines.

There are two details I would like some help correcting (as illustrated in the gif below):
- Indentation of bullet points inside the collapsible content.
- Changing this collapsible background color to white instead of light blue, as validated [here](https://vtex.slack.com/archives/C01KE2JBL5P/p1662118967488599?thread_ts=1661886230.056799&cid=C01KE2JBL5P) with Davis.

Please let me know if there's something I can do to fix that directly in mdx or if I should open an issue in the repository 😄 

collapsible-docs

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 1]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
+<PageSection
+  id={tabToc.designReference[0].id}
+  title={tabToc.designReference[0].value}
+>
+
+<details><summary>When to use the Radio, Switch, or Combobox Multiple components instead of the Checkbox component?</summary>
+<p>
+  <ul>
+    <li>Use the Checkbox component to allow multiple choices, in scenarios where options are not mutually exclusive and you wish to allow one or more selection.</li>
+    <li>Use the Checkbox component to provide binary options, such as yes or no choices, where checking the Checkbox means yes and not checking it means no. Never use a radio component in this scenario.</li>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:554]: Esses dois primeiros pontos podem ser removidos, já estão atendidos pela descrição no overview.

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 2]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
-10. Add descriptions if the control is critical, or complex.
-11. Lists of controls should be vertical for better scannability
-
-👉 For UX Writing guidelines, access https://uxwriting.vtex.com
+<PageSection
+  id={tabToc.designReference[0].id}
+  title={tabToc.designReference[0].value}
+>
+
+<details><summary>When to use the Radio, Switch, or Combobox Multiple components instead of the Checkbox component?</summary>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:550]: Acho que a ordem aqui na pergunta e nas respostas deveria ser: Switch, Radio, Combobox Multiple

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 3]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
+<p>
+  <ul>
+    <li>Use the Checkbox component to allow multiple choices, in scenarios where options are not mutually exclusive and you wish to allow one or more selection.</li>
+    <li>Use the Checkbox component to provide binary options, such as yes or no choices, where checking the Checkbox means yes and not checking it means no. Never use a radio component in this scenario.</li>
+    <li>Choose the Combobox Multiple component if multiple options can be selected and there are more than 5 available options.</li>
+    <li>Choose the Switch component if the action is instant and does not require confirmation. For example, use a switch in a product form to show a new form section only if the product has variants.</li>
+    <li>Choose the Radio component if only one option can be selected at a time.</li>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:557]: Choose the Radio component if only one option can be selected at a time and there are up to five options.

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 4]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
+
+<details><summary>What should be the label of a Checkbox?</summary>
+<p>
+  <ul>
+    <li>Write the label so it corresponds to the value of the Checkbox.</li>
+    <li>Use sentence case but capitalize proper nouns.</li>
+  </ul>
+  <br /><br />
+  What to avoid:
+  <ul>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:625]: Adicionar esse ponto:
Don't write labels that vary a lot in length between options.

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 5]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
+</details>
+
+<details><summary>When should the indeterminate state be used?</summary>
+<p>
+In scenarios where it might be necessary to check or uncheck all options, a Checkbox can be placed in the beginning of the group with the <em>Select all</em> label. Make sure options below are nested and vertically aligned with the <em>Select all</em> label.<br /><br />
+The indeterminate state should be displayed in these scenarios when only a few options (not all) are checked. When clicking the indeterminate state, all options must be checked. When clicking for a second time, all options must be unchecked.
+</p>
+</details>
+
+<details><summary>What should be the default state of a Checkbox?</summary>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:604]: What should be the default value of a Checkbox?

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 6]
```diff
@@ -511,31 +532,157 @@ function Example() {
...
+<p>
+When a Checkbox serves as confirmation, leave it unchecked by default and make it required to direct the attention of merchants. For example, the choice to accept terms and conditions.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:606]: Substituir essa resposta pela resposta que definimos pro Radio.

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 7]
```diff
@@ -118,8 +139,8 @@ export const tabToc = {
 
 ## What is this?
 
-Checkboxes are tools that customize configurations. This control has three states: checked, unchecked and indeterminate. They reflect the selection of an option and its execution usually requires another control, or confirmation.
-
+Checkboxes allow merchants to make multiple independent choices in a form where there are at most five options and each option is binary.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:142]: Checkbox controls allow...

**Review by davicostalf — CHANGES_REQUESTED** [review]
Os pontos que discutimos

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 8]
```diff
@@ -511,31 +532,161 @@ function Example() {
...
+  id={tabToc.designReference[0].id}
+  title={tabToc.designReference[0].value}
+>
+
+<details><summary>When to use the Switch, Radio, or Combobox Multiple components instead of the Checkbox component?</summary>
+<p>
+  <ul>
+    <li>Choose the Switch component if the action is instant and does not require confirmation. For example, use a switch in a product form to show a new form section only if the product has variants.</li>
+    <li>Choose the Radio component if only one option can be selected at a time and there are up to five options.</li>
+    <li>Choose the Combobox Multiple component if multiple options can be selected and there are more than 5 available options.</li>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:555]: "5" should be written as "five"

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 9]
```diff
@@ -511,31 +532,161 @@ function Example() {
...
+</details>
+
+<details><summary>When should the indeterminate state be used?</summary>
+<p>
+In scenarios where it might be necessary to check or uncheck all options, a Checkbox can be placed in the beginning of the group with the <em>Select all</em> label. Make sure options below are nested and vertically aligned with the <em>Select all</em> label.<br /><br />
+The indeterminate state should be displayed in these scenarios when only a few options (not all) are checked. When clicking the indeterminate state, all options must be checked. When clicking for a second time, all options must be unchecked.
+</p>
+</details>
+
+<details><summary>What should be the default value of a Checkbox?</summary>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:602]: Atualizar a resposta com base nos comentários na doc do Radio

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 10]
```diff
@@ -511,31 +532,161 @@ function Example() {
...
+    <li>Choose the Combobox Multiple component if multiple options can be selected and there are more than 5 available options.</li>
+  </ul>
+</p>
+</details>
+
+<details><summary>When to use a Checkbox Group instead of a Checkbox?</summary>
+<p>Since the Checkbox label should only correspond to its value, the choice should be contextualized using the label of a Checkbox Group or the title of a form section. This is why the Checkbox Group is usually the right choice even for binary choices where there is only a single Checkbox.</p>
+</details>
+
+<details><summary>When to use a vertical or a horizontal Checkbox Group?</summary>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:564]: Atualizar a resposta com base nos comentários na doc do Radio

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 11]
```diff
@@ -511,31 +532,161 @@ function Example() {
...
+<details><summary>What should be the help text of a Checkbox?</summary>
+<p>
+  <ul>
+    <li>Include a help text only for critical choices where the consequences of checking the option need to be explained in more detail.</li>
+    <li>Write the help text as if you are completing the following sentence: <em>If you check this option, you or the system will [help text]</em>.</li>
+    <li>Start with verbs in the simple present tense. For example, for the value <em>Fragile</em> the help text should be<em> Prints this information on the shipping label</em> and not <em>Print this information on the shipping label</em>.</li>
+    <li>Use sentence case, but capitalize proper nouns.</li>
+  </ul>
+  <br /><br />
+  What to avoid:
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:664]: <li>Don't include links. Use an alert component if this is necessary.</li>
    <li>Don't write more than one sentence.</li>
    <li>Don't use periods or commas.</li>

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 12]
```diff
@@ -511,31 +532,161 @@ function Example() {
...
+<p>
+  <ul>
+    <li>Write the label so it corresponds to the value of the Checkbox.</li>
+    <li>Use sentence case but capitalize proper nouns.</li>
+  </ul>
+  <br /><br />
+  What to avoid:
+  <ul>
+    <li>Don't write long Checkbox labels. Use the help text or the Checkbox Group label to provide additional context when necessary.</li>
+    <li>Don't write labels that vary a lot in length between options.</li>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:630]: Don't write Checkbox labels that vary a lot in length between options

File: packages/admin-ui-docs/docs/components/checkbox.mdx [hunk 13]
```diff
@@ -511,31 +532,160 @@ function Example() {
...
-👉 For UX Writing guidelines, access https://uxwriting.vtex.com
+<PageSection
+  id={tabToc.designReference[0].id}
+  title={tabToc.designReference[0].value}
+>
+
+<details><summary>When to use the Switch, Radio, or Combobox Multiple components instead of the Checkbox component?</summary>
+<p>
+  <ul>
+    <li>Choose the Switch component if the action is instant and does not require confirmation. For example, use a switch in a product form to show a new form section only if the product has variants.</li>
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/components/checkbox.mdx:553]: Choose the Switch component if the action is instant and does not require confirmation. For example, use a Switch in a User Settings page for the option to <em>Activate notifications<em>.

**Review by davicostalf — CHANGES_REQUESTED** [review]

**Comment by julia-rabello** [comment]
Closing this PR and recreating changes in PR #429 without conflicts as recommended by @lucasaarcoverde.

**Pull request closed without merging**